### DISCUSSION TITLE
New Meters fitted, how to remove old meter entities.

### DISCUSSION THREAD

**Discussion by andyblac** [BottlecapDave/HomeAssistant-OctopusEnergy#370] [Q&A] [answered]
Hi all,

I have just has new meters fitted, how do I remove the old entites for HA, as they do not belong to the recorder I can't rever then by using the reorder purge service, Does any one have any ideas??

<img width="1510" alt="Screenshot 2023-08-12 at 11 22 17" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/77936/558a64b4-e25a-41e2-ba69-67a89af87bc2">

these are what I want removing.

Thanks
Andy.

**Comment by BottlecapDave** [comment, marked as answer]
Ah, because those sensors are not directly tied to the corresponding entities, there's no way currently of deleting them. It also looks like HA prevents you from deleting them unless there's something wrong with them.

I've raised an issue around your problem, but for now it feels like the only way is to manually alter the database (less than ideal)

**Reply by andyblac** [reply]
how do I manually alert the database ?, would that then allow me to delete them?, It not a big issue, just I have to be careful not to select the wrong entity when setting up things like the Energy Dashboard.

Plus, we also can't rename the entities with the domain `octopus_energy` because they do not show up in the entities list, and this results in long name in the Energy Dashboard, is this something you can change ?, maybe shorten them to just `Octopus Electricity XXXXXXXX` and `Octopus GAS XXXXXXXX (kWh)`. XXX being the serial meter number ?

<img width="1219" alt="Screenshot 2023-08-12 at 12 38 51" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/77936/412c02bc-f12e-4cb9-bd3f-cb957ecca8a0">

**Reply by BottlecapDave** [reply]
> how do I manually alert the database ?, would that then allow me to delete them?, It not a big issue, just I have to be careful not to select the wrong entity when setting up things like the Energy Dashboard.

You'll need to install the addon `sqlite web`. Once done, launch the add on, click on `statistics_meta`, click `query` and execute the following query

```sql
SELECT id, statistic_id
FROM "statistics_meta"
WHERE statistic_id LIKE "octopus_energy:%"
```

This will get you a list of all statistics and their ids. Make a note of the ids you want to delete, then replace the previous query with the following query in the same box, placing the list of ids with the ones you want to target and execute. For example, I'm targeting the records whose ids are 141 and 142.

```sql
DELETE FROM "statistics_meta"
WHERE id IN (141, 142);
```

Then replace the previous query with the following query in the same box, placing the list of ids with the ones you want to target and execute.

```sql
DELETE FROM "statistics"
WHERE metadata_id IN (141, 142);
```

This should remove the items from the list. You may need to restart HA to persist the changes. Only do this once you have deleted the old meter sensors, otherwise the integration will add them back while the meters are active.

> Plus, we also can't rename the entities with the domain octopus_energy because they do not show up in the entities list, and this results in long name in the Energy Dashboard, is this something you can change ?, maybe shorten them to just Octopus Electricity XXXXXXXX and Octopus GAS XXXXXXXX (kWh). XXX being the serial meter number ?

I'm hesitant to do that as people might get confused about what the sensors represent (previous vs current) and I believe there are scenarios where serial or mpan/mprn (can't remember which) can be the same across several meters.

These sensors are _hopefully_ a stop gap while I still investigate better ways of associating the data with the sensors themselves. The data appeared correct for the previous day, but then [had weird spikes](https://community.home-assistant.io/t/help-needed-around-importing-historic-statistics/567726) on the day of consumption and I could never get the bottom of it and no one could assist.

**Comment by andyblac** [comment]
i use InfluxDB, would not happen to know the command for this would you, as those SQL query commands don't seen to work.

**Reply by BottlecapDave** [reply]
I'm not sure on those commands I'm afraid

**Comment by BottlecapDave** [comment]
As per https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/issues/371

> I've added a new service which removes custom external statistic ids associated with meters that have no active tariffs.. This can be found in [v8.3.0](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/releases/tag/v8.3.0), and the docs can be found at https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/blob/develop/_docs/setup_account.md#service-octopus_energypurge_invalid_external_statistic_ids

**Comment by andyblac** [comment]
thanks a lot.
